### PR TITLE
Refactor ssl related configurations to improve re-usability

### PR DESCRIPTION
**Pull request by Bhashinee**
## Purpose
Fix the issue: https://github.com/wso2/transport-http/issues/211

### CODE REVIEW

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/config/SslConfiguration.java [hunk 1]
```diff
@@ -108,59 +88,127 @@ public void setParameters(List<Parameter> parameters) {
...
+        if (sslConfig.getKeyStore() == null || sslConfig.getKeyStorePass() == null) {
+            throw new IllegalArgumentException("keyStoreFile or keyStorePassword not defined for HTTPS scheme");
+        }
+        if (!sslConfig.getKeyStore().exists()) {
+            throw new IllegalArgumentException("KeyStore File " + sslConfig.getKeyStore() + " not found");
+        }
+        sslConfig.setCertPass(sslConfig.getKeyStorePass());
+        for (Parameter parameter : parameters) {
+            switch (parameter.getName()) {
+            case SERVER_SUPPORT_CIPHERS:
```
Comment (wggihan) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/config/SslConfiguration.java]: Is this formatting correct?

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/config/SslConfiguration.java [hunk 2]
```diff
@@ -108,59 +88,127 @@ public void setParameters(List<Parameter> parameters) {
...
+
+        sslConfig.setTrustStore(sslConfig.getTrustStore()).setTrustStorePass(sslConfig.getTrustStorePass());
+        String sslProtocol = sslConfig.getSSLProtocol() != null ? sslConfig.getSSLProtocol() : TLS_PROTOCOL;
+        sslConfig.setSSLProtocol(sslProtocol);
+        String tlsStoreType = sslConfig.getTLSStoreType() != null ? sslConfig.getTLSStoreType() : JKS;
+        sslConfig.setTLSStoreType(tlsStoreType);
+        if (parameters != null) {
+            for (Parameter parameter : parameters) {
+                String paramName = parameter.getName();
+                if (CLIENT_SUPPORT_CIPHERS.equals(paramName)) {
```
Comment (wggihan) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/config/SslConfiguration.java]: Can we add a switch statement for this?

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/config/SslConfiguration.java (hunk 1 above)
Comment (Bhashinee) [reply components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/config/SslConfiguration.java]: Fixed the formatting issue.

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/config/SslConfiguration.java (hunk 2 above)
Comment (Bhashinee) [reply components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/config/SslConfiguration.java]: Replaced with switch statements.

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/DefaultHttpWsConnectorFactory.java [hunk 3]
```diff
@@ -68,18 +68,19 @@ public ServerConnector createServerConnector(ServerBootstrapConfiguration server
             ListenerConfiguration listenerConfig) {
         ServerConnectorBootstrap serverConnectorBootstrap = new ServerConnectorBootstrap(allChannels);
         serverConnectorBootstrap.addSocketConfiguration(serverBootstrapConfiguration);
-        serverConnectorBootstrap.addSecurity(listenerConfig.getSSLConfig());
-        serverConnectorBootstrap.addcertificateRevocationVerifier(listenerConfig.validateCertEnabled());
-        serverConnectorBootstrap.addCacheDelay(listenerConfig.getCacheValidityPeriod());
-        serverConnectorBootstrap.addCacheSize(listenerConfig.getCacheSize());
-        serverConnectorBootstrap.addOcspStapling(listenerConfig.isOcspStaplingEnabled());
+        serverConnectorBootstrap.addSecurity(listenerConfig.getListenerSSLConfig());
+        if (listenerConfig.getListenerSSLConfig() != null) {
```
Comment (chamil321) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/DefaultHttpWsConnectorFactory.java]: listenerConfig.getListenerSSLConfig() seems redundant. Shall we create a local variable and reuse it?

Comment (Bhashinee) [reply components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/DefaultHttpWsConnectorFactory.java]: Fixed it.

**Pull request merged**